### PR TITLE
React deployment on GH Pages

### LINKED ISSUE #5: Setup GH Pages to deploy Vite React app
Follow instructions [here](https://vite.dev/guide/static-deploy#github-pages).

This is blocked by #2. Either wait for that to be merged or branch off of that feature branch.

### PR DESCRIPTION
**Pull request by ashstewart7**
Closes #5 

Adds a GH Actions workflow (almost entirely copy pasted from Vite docs) to deploy to GH pages on commits to main.

### CODE REVIEW

File: .github/workflows/pages-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
```
Comment (aedwardg) [inline .github/workflows/pages-deploy.yml]: Do you think we can update this to be 23 and also update this line over in the other action?
https://github.com/CodeCafeCommunity/codecafecommunity.github.io/blob/f1821a016811d2197cd683b1020998ecca4853a3/.github/workflows/format-and-lint.yml#L14

I'd like us to use the latest version where we can.
I will also open up another ticket to have a `.nvmrc` or `.node-version` file that specifies version 23

**Review by aedwardg — COMMENTED** [review]
Looks good, just one comment

File: .github/workflows/pages-deploy.yml (hunk 1 above)
Comment (aedwardg) [reply .github/workflows/pages-deploy.yml]: Actually, I took care of the one in `format-and-lint.yml` in #13, so no need to change that file.

**Review by Victoria-DR — APPROVED** [review]
LGTM

File: .github/workflows/pages-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          cache: "npm"
+      - name: Install dependencies
+        run: npm ci
+      - name: Build
+        run: npm run build
```
Comment (aedwardg) [inline .github/workflows/pages-deploy.yml:39]: Actually it looks like we still need to add this build script 
https://vite.dev/guide/static-deploy

**Review by aedwardg — CHANGES_REQUESTED** [review]

File: .github/workflows/pages-deploy.yml (hunk 2 above)
Comment (Jamesllllllllll) [reply .github/workflows/pages-deploy.yml:39]: I think this looks fine, it's following the GitHub Pages example. `npm run build` will run the vite build script and deploy, no?

Comment (aedwardg) [reply .github/workflows/pages-deploy.yml:39]: No, there is no build script in the package.json yet. It is missing the following code from the docs
```
{
  "scripts": {
    "build": "vite build",
    "preview": "vite preview"
  }
}
```

Comment (aedwardg) [reply .github/workflows/pages-deploy.yml:39]: Nevermind, looks like I completely missed them and they are already included. Disregard.

**Review by aedwardg — APPROVED** [review]

**Pull request merged**